doc/radosgw/d3n_datacache.rst

```diff
@@ -0,0 +1,107 @@
+
+Limitations
+-----------
+
+- D3N will not cache objects compressed by `Rados Gateway Compression`_ (OSD level compression is supported).
+- D3N will not cache objects encrypted by `Rados Gateway Encryption`_.
+- D3N will be disabled if the `rgw_max_chunk_size` config variable value differs from the `rgw_obj_stripe_size` config variable value.
+
+
+D3N Environment Setup
+=====================
+
+Running
+-------
+
+D3N minimal configuration startup example using `vstart.sh`::
+
+    env MON=1 OSD=1 MDS=0 MGR=1 RGW=1 ../src/vstart.sh -n  \
+    -o rgw_d3n_l1_local_datacache_enabled=true  \
+    -o rgw_d3n_l1_datacache_persistent_path=/mnt/nvme0/rgw_datacache/  \
+    -o rgw_d3n_l1_datacache_size=10737418240
+
+the above example assumes that the cache backing-store solid state device
+is mounted at `/mnt/nvme0` and has `10 GB` of free space available.
+
+    NOTE: Each time the Rados Gateway is restarted the content of the cache directory is purged.
+
+Logs
+----
+- D3N related log lines in `radosgw.*.log` contain the string `d3n` (case insensitive).
+- low level D3N logs can be enabled by the `debug_rgw_datacache` subsystem (up to `debug_rgw_datacache=30`)
+
+
+Configuration
+=============
+
+D3N related `ceph.conf` configuration variables:
+
++----------------------------------------------------+----------+------------------------------------------------------------------------------------------------+
+| Configuration Variable                             | Type     | Description                                                                                    |
++----------------------------------------------------+----------+------------------------------------------------------------------------------------------------+
+| ``rgw_d3n_l1_local_datacache_enabled``             | bool     | Enable datacenter-scale dataset delivery local cache                                           |
++----------------------------------------------------+----------+------------------------------------------------------------------------------------------------+
+| ``rgw_d3n_l1_datacache_persistent_path``           | string   | path of the directory for storing the local cache objects data                                 |
++----------------------------------------------------+----------+------------------------------------------------------------------------------------------------+
+| ``rgw_d3n_l1_datacache_size``                      | size     | datacache maximum size on disk in bytes                                                        |
++----------------------------------------------------+----------+------------------------------------------------------------------------------------------------+
+| ``rgw_d3n_l1_eviction_policy``                     | enum     | select the d3n cache eviction policy, allowed values: (`lru`|`random`)                         |
```

Comment:
i think we should prefer the yaml integration for these, so these docs are automatically updated to match the source:
```rst
.. confval:: rgw_d3n_l1_local_datacache_enabled
.. confval:: rgw_d3n_l1_datacache_persistent_path
.. confval:: rgw_d3n_l1_datacache_size
.. confval:: rgw_d3n_l1_eviction_policy
```
for an example of how these render, see https://docs.ceph.com/en/latest/radosgw/config-ref/

Reply:
very cool, done, thanks